src/system.c

```diff
@@ -123,7 +123,26 @@ void sys_set_seccomp_syscall(bool enable)
  */
 int sys_chk_seccomp_action(uint32_t action)
 {
-	if (action == SCMP_ACT_KILL) {
+	int rc, nr_seccomp;
+	uint32_t kaction = action & SECCOMP_RET_ACTION_FULL;
+
+
+	nr_seccomp = arch_syscall_resolve_name(arch_def_native, "seccomp");
+	if (nr_seccomp != __NR_SCMP_ERROR) {
+		rc = syscall(nr_seccomp, SECCOMP_GET_ACTION_AVAIL, 0,
+			     &kaction);
+		if (rc == -EPERM)
```

Comment:
I'm not fond of this "if", but I couldn't figure out a more graceful solution.  We really should ask the kernel if the requested action is supported here, but in Valgrind's case, the seccomp syscall [isn't supported](http://repo.or.cz/valgrind.git/blob/6ae2edea014669d8082747f0f268e9404e0fd296:/coregrind/m_syswrap/syswrap-amd64-linux.c#l840).

I tried using sys_chk_seccomp_syscall(), but that will fail in a few tests because _support_seccomp_syscall gets set to 1 when the api level is set.  But, _nr_seccomp isn't set using that interface and remains -1.  Test 06-sim-actions.c exhibits this behavior.

Reply:
Kees had also mentioned using `/proc/sys/kernel/seccomp/actions_avail` as another avenue to get the available actions.  But that produces a list like the following - `kill_process kill_thread trap errno trace log allow
`, and I wasn't keen on doing string processing.

Reply:
I would look at what we did for SCMP_ACT_LOG and duplicate that, I believe it should do everything you need.